xarray/core/dataset.py

```diff
@@ -2384,6 +2384,34 @@ def from_dataframe(cls, dataframe):
             obj[name] = (dims, data)
         return obj
 
+    def to_dask_dataframe(self, set_index=True):
+        """
+        Convert this dataset into a dask.dataframe.DataFrame.
+
+        Non-index variables in this dataset form the columns of the
+        DataFrame.
+
+        If set_index=True, the dask DataFrame is indexed by
+        this dataset's coordinate. Since dask DataFrames to not support
+        multi-indexes, this only works if there is one coordinate dimension.
+        """
+
+        import dask.dataframe as dd
+        import dask.array as da
+
+        columns = [k for k in self if k not in self.dims]
+
+        data = [self._variables[k].data.reshape(-1) for k in columns]
+        df = dd.from_dask_array(da.stack(data, axis=1), columns=columns)
```

Comment:
Stacking together variables into a single dask array loses dtype information, which is not ideal. Can you look into constructing a dask dataframe from a list of dask arrays?